api/api/src/main/scala/uk/ac/wellcome/platform/api/services/ImagesService.scala

```diff
@@ -31,4 +45,42 @@ class ImagesService(searchService: ElasticsearchService)(
         }
       }
 
+  def listOrSearchImages(index: Index, searchOptions: ImagesSearchOptions)
+    : Future[Either[ElasticError, ResultList[AugmentedImage, Unit]]] =
+    searchService
+      .executeSearch(
+        queryOptions = toElasticsearchQueryOptions(searchOptions),
+        index = index,
+        scored = searchOptions.searchQuery.isDefined
```

Comment:
I wonder if this could just be calculated when we create the searchOptions, keeping it at the edge e.g. (bad code)
```
val searchOptions = SearchOptions(
  // ...
  sort = queryParams.sort.getOrElse(params.query match {
   case Some(a) => ScoreSort
   case None => IdSort
  })
)
```

That way we could also sort on a non-queried list. e.g. `images?sort=colour`.

Reply:
Hmmm, I don't really want to "lie" about where the search options are from - they should reflect what the user asked for and no more. 

This is only a refactor of the old `listOrSearchWorks` code. Regardless of query, we always sort by ID, and when there's a query we sort by score as well - that's why those sorts live inside the request builder. 

There's nothing stopping us sorting on a list other than that we don't have anything to sort on yet - you can see an example of us doing that in the works request builder.

Reply:
Hmm. I thought we had the `SearchParams` which then mapped to `SearchOptions`. page is a good example of this where `Option[String] => Int`.

Yea, it's something that has bothered from the `listOrSearch` method.

I suppose my main annoyance is it's not closer to the edge, either the Elastic client of the Request client, but rather sitting in this service in the middle.

> other than that we don't have anything to sort on yet

With works we already have date search, which _should_ work on search or list.

Reply:
> With works we already have date search, which should work on search or list.

We already do that, no?

Reply:
[Yip](https://github.com/wellcomecollection/catalogue/pull/581/files#diff-6068495eb1eb5651cc269501889f73b8R26-R30). I was just looking at the image implementation that didn't allow for that (rightfully as we don't have another sort).

The main objection again is having three places where sort is configured, instead of potentially 2 or 1.

Reply:
In essence maybe moving the scored into the `executeSearch` method.

Having an argument called `scored` defined by if there is a `query` that affects `sort` is a little hard to figure out at first glance.

Reply:
Yeah, agreed - had wanted it to be explicit to the caller that there was some extra stuff going on but to be honest it's slightly surprising that you have to ask to sort by score anyway